### DISCUSSION TITLE
Best way for state rehydration

### DISCUSSION THREAD

**Discussion by rendomnet** [vuejs/pinia#1946] [Help and Questions]
We need from time to time rehydrate(replace) our state. Because we can have several apps running in the separated context but they all needs to have same state.
We are recieving new state in json format.

What is a best way to replace current state with new one? So equal values in state would not trigger reactivity?

**Comment by posva** [comment]
If you want to avoid triggering reactivity, [you shall replace the state **before the stores are instantiated**](https://pinia.vuejs.org/ssr/index.html#state-hydration) (aka the first time `useSomeStore()` is called):

```ts
pinia.state.value = {
  someStoreId: {...}
}
```

**Reply by rendomnet** [reply]
@posva No in our case we need to rehydrate the state regularly. 
App is working in several tabs simultaneously.
So for example api got new STATE object and we need to replace current state with new one(or replace only specific state child nodes).

**Reply by posva** [reply]
In that case you **do want to trigger reactivity** 🙃
You can try using `toRaw()` in some places (Vue 3 only) to manualyl opt out of reactivity: https://vuejs.org/api/reactivity-advanced.html#toraw
